packages/@dynamic-labs-connectors/sequence-cross-app-evm/src/SequenceWaasTransportProvider.ts

```diff
@@ -0,0 +1,168 @@
+          throw e;
+        }
+      }
+      const account = getAddress(walletAddress);
+      return [account];
+    }
+
+    if (method === 'wallet_switchEthereumChain') {
+      const chainId = normalizeChainId(params?.[0].chainId);
+
+      const networkName = allNetworks.find((n) => n.chainId === chainId)?.name;
+      const jsonRpcProvider = new ethers.JsonRpcProvider(
+        `${this.nodesUrl}/${networkName}/${this.projectAccessKey}`,
+      );
+
+      this.jsonRpcProvider = jsonRpcProvider;
+      this.currentNetwork = ethers.Network.from(chainId);
+
+      return null;
+    }
+
+    if (method === 'eth_chainId') {
+      return ethers.toQuantity(this.currentNetwork.chainId);
+    }
+
+    if (method === 'eth_accounts') {
+      const address = this.transport.getWalletAddress();
+      if (!address) {
+        return [];
+      }
+      const account = getAddress(address);
+      return [account];
+    }
+
+    if (method === 'eth_sendTransaction') {
+      if (!params) {
+        throw new Error('No params');
+      }
+
+      try {
+        const response = await this.transport.sendRequest(
+          method,
+          params,
+          this.getChainId(),
+        );
+
+        if (response.code === 'transactionFailed') {
+          // Failed
+          throw new TransactionRejectedRpcError(
+            new Error(`Unable to send transaction: ${response.data.error}`),
+          );
+        }
+
+        if (response.code === 'transactionReceipt') {
+          // Success
+          const { txHash } = response.data;
+          return txHash;
+        }
+      } catch (e) {
+        console.log('error in sendTransaction', e);
+        throw new TransactionRejectedRpcError(
+          new Error(`Unable to send transaction: wallet window was closed.`),
+        );
+      }
+    }
+
+    if (
+      method === 'eth_sign' ||
+      method === 'eth_signTypedData' ||
+      method === 'eth_signTypedData_v4' ||
+      method === 'personal_sign'
+    ) {
+      if (!params) {
+        throw new Error('No params');
+      }
+      try {
+        const response = await this.transport.sendRequest(
+          method,
+          params,
+          this.getChainId(),
+        );
+
+        return response.data.signature;
+      } catch (e) {
+        console.log('error in sign', e);
+        throw new TransactionRejectedRpcError(
+          new Error(`Unable to sign: wallet window was closed.`),
+        );
+      }
+    }
+
+    return await this.jsonRpcProvider.send(method, params ?? []);
+  }
+
+  override async getTransaction(txHash: string) {
+    return await this.jsonRpcProvider.getTransaction(txHash);
+  }
+
+  detectNetwork(): Promise<ethers.Network> {
+    return Promise.resolve(this.currentNetwork);
+  }
+
+  getChainId() {
+    return Number(this.currentNetwork.chainId);
+  }
+
+  disconnect() {
+    this.transport.disconnect();
+  }
+}
+
+function normalizeChainId(
+  chainId: string | number | bigint | { chainId: string },
```

Comment:
this function was moved to the `utils` file, so we should remove it from this one